src/lavinmq/message_store.cr

```diff
@@ -472,8 +474,9 @@ module LavinMQ
       if deleted = @deleted[seg]?
         deleted.each do |pos|
           mfile.pos = pos
-          bytesize -= BytesMessage.skip(mfile)
-          count -= 1
+          bs = BytesMessage.skip(mfile)
+          bytesize -= bs if bytesize >= bs # Don't allow underflow
+          count -= 1 if count > 0          # Don't allow underflow
```

Comment:
This means we maybe won't reach `bytesize = 0`?
Would it be better to do `@bytesize = {0, bytesize}.max` instead of the if-statement here? And if so, maybe do the same for `count` for consistency.

Reply:
Yeah, makes sense. Will change!

Reply:
Hmm, no, maybe it doesn't make sense 🤔 
bytesize is a UInt, so we will just get an ArithmeticOverflow if we try to decrease bytesize further than 0. 

Maybe something like this instead? 
```
bytesize = bs < bytesize ? bytesize - bs : 0
count -= 1 if count > 0
```